pkg/users/group-create-dialog.js

```diff
@@ -0,0 +1,164 @@
+
+import { has_errors, is_valid_char_name } from "./dialog-utils.js";
+
+const _ = cockpit.gettext;
+
+function GroupCreateBody({ state, errors, change }) {
+    const {
+        name, id,
+    } = state;
+
+    return (
+        <Form isHorizontal onSubmit={apply_modal_dialog}>
+            <FormGroup label={_("Name")}
+                       helperTextInvalid={errors && errors.name}
+                       validated={(errors && errors.name) ? "error" : "default"}
+                       fieldId="groups-create-name">
+                <TextInput id="groups-create-name"
+                           validated={(errors && errors.name) ? "error" : "default"}
+                           value={name} onChange={value => change("name", value)} />
+            </FormGroup>
+
+            <FormGroup label={_("ID")}
+                       hasNoPaddingTop
+                       helperTextInvalid={errors && errors.id}
+                       validated={(errors && errors.id) ? "error" : "default"}
+                       isStack
+                       fieldId="groups-create-id">
+                <TextInput id="groups-create-id"
+                           validated={(errors && errors.id) ? "error" : "default"}
+                           value={id} onChange={value => change("id", value)} />
+            </FormGroup>
+        </Form>
+    );
+}
+
+function validate_name(name, groups) {
+    if (!name)
+        return _("No group name specified");
+
+    for (let i = 0; i < name.length; i++) {
+        if (!is_valid_char_name(name[i]))
+            return _("The group name can only consist of letters from a-z, digits, dots, dashes and underscores");
+    }
+
+    for (let k = 0; k < groups.length; k++) {
+        if (groups[k].name == name)
+            return _("A group with this name already exists");
+    }
+
+    return null;
+}
+
+function validate_group(id, groups) {
+    if (!id)
+        return _("No ID specified");
+
+    const id_num = parseInt(id);
+    if (id_num.toString() !== id || id_num < 0)
+        return _("The group ID must be positive integer");
+
+    return null;
+}
+
+export function group_create_dialog(groups, setGroupsCardExpanded) {
+    //  Handle also the case where logindef == null, i.e. the file does not exist.
+    //  While that's unusual, "empty /etc" is a goal, and it shouldn't crash the page.
+    cockpit.file("/etc/login.defs").read()
+            .then(logindef => {
+                const gids = groups
+                        .filter(g => g.name !== 'nobody')
+                        .map(group => group.gid);
+                const min_gid = parseInt(logindef.match(/^GID_MIN\s+(\d+)/m)[1] || 500);
+                const max_gid = parseInt(logindef.match(/^GID_MAX\s+(\d+)/m)[1] || 60000);
+
+                change("id", (Math.max(min_gid, Math.max(...gids.filter(id => id < max_gid)) + 1) + 1).toString());
+            }, () => Promise.resolve());
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18136-20230125-165802-b16c2652-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)